src/distributions/adopt/installer.ts

```diff
@@ -7,13 +7,20 @@ import semver from 'semver';
 
 import { JavaBase } from '../base-installer';
 import { IAdoptAvailableVersions } from './models';
-import { JavaInstallerOptions, JavaDownloadRelease, JavaInstallerResults } from '../base-models';
-import { MACOS_JAVA_CONTENT_POSTFIX } from '../../constants';
+import { JavaDownloadRelease, JavaInstallerOptions, JavaInstallerResults } from '../base-models';
 import { extractJdkFile, getDownloadArchiveExtension, isVersionSatisfies } from '../../util';
 
+export enum AdoptImplementation {
+  Hotspot = 'Hotspot',
+  OpenJ9 = 'OpenJ9'
+}
+
 export class AdoptDistribution extends JavaBase {
-  constructor(installerOptions: JavaInstallerOptions) {
-    super('Adopt', installerOptions);
+  private jvmImpl: string;
+
+  constructor(installerOptions: JavaInstallerOptions, jvmImpl: AdoptImplementation) {
+    super(`Adopt-${jvmImpl}`, installerOptions);
```

Comment:
There is a one problem with these changes. After these changes, Java will be located into `Java_Adopt-Hotspot_jdk` and `Java_Adopt-OpenJ9_jdk` folders in toolcache. See https://github.com/actions/setup-java/blob/main/src/distributions/base-installer.ts#L66

The problem is that Hosted images already have Adopt HotSpot cached on images in folder `Java_Adopt_jdk` so after these changes, cached versions won't be found.

I think we can fix it by overriding `toolcacheFolderName` in `AdoptDistribution` class.
Something like that:
```ts
protected get toolcacheFolderName(): string {
    if (this.jvmImpl === AdoptImplementation.Hotspot) {
      // exclude Hotspot postfix from distribution name because Hosted runners have pre-cached Adopt OpenJDK under "Java_Adopt_jdk"
      return `Java_Adopt_${this.packageType}`;
    }
    return super.toolcacheFolderName()
}
```